File: python/3142. Check if Grid Satisfies Conditions.py
class Solution:
    def satisfiesConditions(self, grid: List[List[int]]) -> bool:
        for j in range(len(grid[0])):
            if j != 0 and grid[0][j] == grid[0][j-1]:
                return False
            for i in range(len(grid)):
                if i != 0 and grid[i][j] != grid[i-1][j]:
                    return False
        return True

File: go/3142. Check if Grid Satisfies Conditions.go
func satisfiesConditions(grid [][]int) bool {
	for j := 0; j < len(grid[0]); j++ {
		if j != 0 && grid[0][j] == grid[0][j-1] {
			return false
		}
		for i := 0; i < len(grid); i++ {
			if i != 0 && grid[i][j] != grid[i-1][j] {
				return false
			}
		}
	}
	return true
}